pkg/sync/adapter.go

```diff
@@ -0,0 +1,222 @@
+package sync
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/openshift/odo/pkg/devfile/adapters/common"
+	versionsCommon "github.com/openshift/odo/pkg/devfile/parser/data/common"
+	"github.com/openshift/odo/pkg/exec"
+	"github.com/openshift/odo/pkg/kclient"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/util"
+
+	"github.com/pkg/errors"
+)
+
+// New instantiantes a component adapter
+func New(adapterContext common.AdapterContext, client SyncClient) Adapter {
+	return Adapter{
+		Client:         client,
+		AdapterContext: adapterContext,
+	}
+}
+
+// Adapter is a component adapter implementation for sync
+type Adapter struct {
+	Client SyncClient
+	common.AdapterContext
+}
+
+// SyncFiles checks whether files have changed in a project
```

Comment:
The function description should be updated, as it doesn't just check if the files changed:
- If files are passed in from watch, it syncs those files directly
- Otherwise, it determines which files changed and then syncs them to the conainer